### PR TITLE
6.0 pre

### PR DESCRIPTION
**Pull request by bobbycyiii**
#59 I changed ``hasSplittingSurfaces()`` in ``engine/triangulation/surfaces.cpp`` to avoid a vertex enumeration.

#60 I changed ``CMakeLists.txt`` so that running ``cmake -DREQUIRE_PYTHON_VERSION=3 ..`` compiles Regina with my Python3 libraries.

### CODE REVIEW

**Comment by WPettersson** [comment]
This is a bit to review, so may take a bit, but for starters, this won't be accepted into master until we (@baburton) are ready for a 6.0 release. If you want to make a PR into my 6.0_pre branch instead, that's more likely to be accepted.

File: engine/triangulation/dim3/surfaces.cpp [hunk 1]
```diff
@@ -237,35 +237,85 @@ bool Triangulation<3>::isZeroEfficient() {
...
+                    Edge<3>* opp = tet_e->edge(v_perm[2],v_perm[3]);
+                    if (!disjoint.count(opp)) candidate_disjoint.push_front(opp);
+                }
+            }
         }
+        if (candidate_disjoint.empty())
+            // We didn't break without removing an edge.
+            // So we must have partitioned the edges into disjoint and intersecting.
+            // Thus there is a splitting surface.
+            return true;
```
Comment (WPettersson) [inline engine/triangulation/dim3/surfaces.cpp]: Is there a reason for not setting splittingSurface_ = true here?

Comment (bobbycyiii) [reply engine/triangulation/dim3/surfaces.cpp]: I did set ``splittingSurface_ = true`` there; looks like the pull request is outdated for some reason. However, I made that mistake earlier in the routine; I'll fix it.

Comment (bobbycyiii) [reply engine/triangulation/dim3/surfaces.cpp]: Feel free to close this PR; I'll submit the next one to the 6.0_pre branch.

**Pull request closed without merging**